### PR TITLE
Fix x-on in x-for when updating

### LINKED ISSUE #413: Events stop firing when I "change a page"
This is not easy to describe so I've created a small jsfiddle with the required code to reproduce the problem:

https://jsfiddle.net/yn4h37pf/

What happens is that I want to implement pagination in a list of items and have each of these items respond to a click event. When you click on the numbers you'll see the event fire in the console log. However if you press "Next page" to load the next page you'll see that although it displays the next 3 numbers, when you click on their buttons nothing will happen. Even if you press previous page to return to the 1st page you still won't get any output in the console.

Am I doing something wrong or maybe this is something not supported at all or this is a bug ?

TIA !

### PR DESCRIPTION
**Pull request by SimoTod**
Fixes #413 

When we update a x-for, if there is a new item we are wrongly calling update instead of insert. More details in the issue.

### CODE REVIEW

File: dist/alpine-ie11.js [hunk 1]
```diff
@@ -5911,7 +5906,9 @@
   }
 
   function lookAheadForMatchingKeyedElementAndMoveItIfFound(nextEl, currentKey) {
-    // If the the key's DO match, no need to look ahead.
+    // We don't know if nextEl is avtually an element so we check that it's not a falsy value first
```
Comment (muzafferdede) [inline dist/alpine-ie11.js:5909]: // We don't know if nextEl is ~~avtually~~ actually an element so we check that it's not a falsy value first

File: dist/alpine.js [hunk 2]
```diff
@@ -486,7 +481,9 @@
   }
 
   function lookAheadForMatchingKeyedElementAndMoveItIfFound(nextEl, currentKey) {
-    // If the the key's DO match, no need to look ahead.
+    // We don't know if nextEl is avtually an element so we check that it's not a falsy value first
```
Comment (muzafferdede) [inline dist/alpine.js:484]: // We don't know if nextEl is ~~avtually~~ actually an element so we check that it's not a falsy value first

File: src/directives/for.js [hunk 3]
```diff
@@ -114,6 +109,9 @@ function addElementInLoopAfterCurrentEl(templateEl, currentEl) {
 }
 
 function lookAheadForMatchingKeyedElementAndMoveItIfFound(nextEl, currentKey) {
+    // We don't know if nextEl is avtually an element so we check that it's not a falsy value first
```
Comment (muzafferdede) [inline src/directives/for.js:112]: // We don't know if nextEl is ~~avtually~~ actually an element so we check that it's not a falsy value first

Comment (SimoTod) [reply src/directives/for.js]: right, I can't spell words correctly, lol. Thanks.

**Comment by muzafferdede** [comment]
@SimoTod Nice one! I am afraid say "Oh Yeah!". Because this `x-for` has been buggy since forever. I strongly believe it will be  re-factored again.  Probably what is happing with V3 already.

File: test/for.spec.js [hunk 4]
```diff
@@ -448,3 +448,62 @@ test('nested x-for event listeners', async () => {
         expect(document._alerts[2]).toEqual('foo: bob = 2')
     })
 })
+
+
+test('listener are bound correctyl on update', async () => {
```
Comment (HugoDF) [inline test/for.spec.js:453]: typos
```suggestion
test('listeners are bound correctly on update', async () => {
```

Comment (SimoTod) [reply test/for.spec.js]: Thanks

**Comment by calebporzio** [comment]
Thanks @SimoTod for the fix, that was a tricky one - great work!

**Pull request merged**